ipfsconn/ipfshttp/ipfshttp_test.go

```diff
@@ -287,6 +287,22 @@ func TestRepoStat(t *testing.T) {
 	}
 }
 
+func TestResolve(t *testing.T) {
+	ipfs, mock := testIPFSConnector(t)
+	defer mock.Close()
+	defer ipfs.Shutdown()
+	time.Sleep(2 * time.Second)
```

Comment:
Why sleep?

Reply:
Without this sometimes context would cancel while connecting to swarm, this doesn't fail the test though.
```
$ go test -v github.com/ipfs/ipfs-cluster/ipfsconn/ipfshttp --run TestResolve
=== RUN   TestResolve
14:17:54.458 ERROR   ipfshttp: error posting to IPFS: Post http://127.0.0.1:44907/api/v0/swarm/connect?arg=/ip4/127.0.0.1/tcp/4001/ipfs/QmXZrtE5jQwXNqCJMfHUTQkvhQ4ZAnqMnmzFMJfLewuabc: context canceled ipfshttp.go:398
--- PASS: TestResolve (0.00s)
PASS
ok  	github.com/ipfs/ipfs-cluster/ipfsconn/ipfshttp	0.020s
```
Nothing special about 2 second, 1 sec works too.

Reply:
I think this one is also there for the same reason, waiting for more than `ConnectSwarmsDelay`(0)
https://github.com/ipfs/ipfs-cluster/blob/a244af9a13558d59ac3670e266516fa2eb95b26f/ipfsconn/ipfshttp/ipfshttp_test.go#L178-L188

Reply:
Left a new comment. The logged error does not affect the test. No need to sleep (faster tests always better).